source/includes/_ethereum.md

```diff
@@ -725,30 +902,148 @@ The random datasource is currently available on the Ethereum mainnet and on all
 The `oraclize_newRandomDSQuery` can be used for different kind of interactions, but the security can be incresed further by additing additional commitment data to the request. For example, for two party interactions, the `oraclize_newRandomDSQuery` can be modified as showon the side to include the sender address and the value send along as commitment data. This more strongly commitment the request for random bytes to current party, which are assumed to have a stake in the contract, making it impossible for miners to replay transactions on potential forks or reorg of the current chain.
 
 #### Multi-Party Interactions
+
 In the case of multi-party interactions, such as voting schemes or lotteries, the commitment data can should include all participants addresses, to ensure that the transaction cannot be replayed by a miner on a fork or a reorged chain where a participant didn't put a stake.
 
+### ERC20 Token Payments
+
+```javascript
 
+/**
+ *
+ * @dev Provable helper functions for managing ERC20 Payments
+ *
+ */
 
-### ProofShield
+// Set payment method to token existing at <token-address>
+oraclize_setCustomTokenPayment(address <token-address>);
 
-The Oraclize *ProofShield* is a concept first introduct at Devcon4, you can watch our presentation about "Scalable Onchain Verification for Authenticated Data Feeds and Offchain Computations" [here](https://www.youtube.com/watch?v=7uQdEBVu8Sk).
+// Approve an allowance of the token for the Provable service:
+oraclize_approveTokenAllowance(
+  address <token-address>,
+  uint256 <token-amount>
+);
+
+// Or combine the above into a single call to this function:
+oraclize_setAndApproveTokenAllowance(
+ 	address<token-address>,
+	uint256<token-amount>
+);
+
+// Get the price of a query in tokens:
+oraclize_getPriceERC20(uint256 <gas-limit>);
+
+// Revert back to paying with ETH and revoke the approved token amount:
+oraclize_unsetAndRevokeCustomPayment();
+
+```
+
+Provable supports ERC20 token payments. In order to make queries using a token as payment rather than ETH, you must somewhere in your contract call: `oraclize_setCustomTokenPayment(address <token-address>);`, where `<token-address>` is any of those existing on the whitelist<!-- FIME Link here! --> of supported tokens.
```

Comment:
FIXME needs fixing

maybe platform info is useful for this?

Reply:
Yeah, perfect candidate for `platform-info`. Given that a whitelist currently doesn't exist yet, I'll leave it as `coming-soon` or something in the docs.